demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2273,15 +2310,89 @@ export class DemoMeetingApp
     this.log(`session stopped from ${JSON.stringify(sessionStatus)}`);
     this.log(`resetting stats in WebRTCStatsCollector`);
     this.statsCollector.resetStats();
+
+    const returnToStart = () => {
+      switch (this.behaviorAfterLeave) {
+        case 'spa':
+          this.switchToFlow('flow-authenticate');
+          break;
+        case 'reload':
+          window.location.href = window.location.pathname;
+          break;
+        // This is useful for testing memory leaks.
+        case 'halt': {
+          // Wait a moment to make sure cleanup is done.
+          setTimeout(() => {
+            // Kill all references to code and content.
+            // @ts-ignore
+            window.app = undefined;
+            // @ts-ignore
+            window.app_meetingV2 = undefined;
+            // @ts-ignore
+            window.webpackHotUpdateapp_meetingV2 = undefined;
+            document.getElementsByTagName('body')[0].innerHTML = '<b>Gone</b>';
+            this.removeFatalHandlers();
+          }, 2000);
+          break;
+        }
+      }
+    };
+
+    /**
+     * This is approximately the inverse of the initialization method above.
+     * This work only needs to be done if you want to continue using the page; if
+     * your app navigates away or closes the tab when done, you can let the browser
+     * clean up.
+     */
+    const cleanUpResources = async () => {
+      // Clean up the timers for this.
+      this.audioVideo.unsubscribeFromActiveSpeakerDetector(this.activeSpeakerHandler);
+
+      // Stop listening to attendee presence.
+      this.audioVideo.realtimeUnsubscribeToAttendeeIdPresence(this.attendeeIdPresenceHandler);
+
+      // Stop watching device changes in the UI.
+      this.audioVideo.removeDeviceChangeObserver(this);
+
+      // Stop content share and local video.
+      await this.audioVideo.stopLocalVideoTile();
+      await this.audioVideo.stopContentShare();
+
+      // Drop the audio output.
+      await this.audioVideo.chooseAudioOutputDevice(null);
```

Comment:
Don't we want to also chooseAudioInput and VideoInput to null? Or can we call audioVideo.deviceController.destroy here?

Reply:
We discussed this in earlier versions.

See three lines below your comment: leaving the meeting will do this for inputs.